ci/builder/Dockerfile

```diff
@@ -10,7 +10,66 @@
 # Build a cross-compiling toolchain that targets the oldest version of Linux
 # that we support.
 
-FROM ubuntu:jammy-20221130 AS crosstool
+FROM debian:bookworm-20230814 as crosstool
+ARG ARCH_GCC
+ARG ARCH_GO
+
+WORKDIR /scratch
+
+# The environment variables are necessary to convince `tzdata`'s install scripts
+# not to hang.
+# See: https://serverfault.com/questions/949991/how-to-install-tzdata-on-a-ubuntu-docker-image
+RUN apt-get update && TZ=UTC DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    autoconf \
+    automake \
+    bison \
+    bzip2 \
+    ca-certificates \
+    curl \
+    file \
+    flex \
+    gawk \
+    g++ \
+    gcc \
+    gnupg2 \
+    help2man \
+    libc-dev \
+    libncurses-dev \
+    libtool-bin \
+    make \
+    patch \
+    texinfo \
+    unzip \
+    xz-utils
+
+COPY crosstool.asc ./
+
+RUN gpg --import crosstool.asc \
+    && rm crosstool.asc \
+    && echo "trusted-key 09f6dd5f1f30ef2e" >> ~/.gnupg/gpg.conf \
+    && mkdir crosstool \
+    # TODO(teske): Remove the CT_ZLIB_VERSION workaround below when we upgrade crosstools. \
+    && curl -fsSL http://crosstool-ng.org/download/crosstool-ng/crosstool-ng-1.25.0.tar.xz > crosstool.tar.xz \
+    && curl -fsSL http://crosstool-ng.org/download/crosstool-ng/crosstool-ng-1.25.0.tar.xz.sig > crosstool.sig \
+    && gpg --verify crosstool.sig crosstool.tar.xz \
+    && tar -xf crosstool.tar.xz -C crosstool --strip-components=1 \
+    && rm crosstool.sig crosstool.tar.xz \
+    && (cd crosstool && ./configure && make install) \
+    && rm -rf crosstool
+
+COPY crosstool-$ARCH_GCC.defconfig ./
+
+RUN DEFCONFIG=crosstool-$ARCH_GCC.defconfig ct-ng defconfig \
+    && rm crosstool-$ARCH_GCC.defconfig \
+    # https://github.com/crosstool-ng/crosstool-ng/issues/1832 \
+    && sed -i 's/CT_ZLIB_VERSION="1.2.12"/CT_ZLIB_VERSION="1.3"/g' .config \
+    && ct-ng build
```

Comment:
What's going on here? This all seems duplicated.

Reply:
Possibly a bad rebase! Will fix!